src/Filter/EmptyFieldFilter.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Filter;
+
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Form\Type\Filter\DefaultType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+final class EmptyFieldFilter extends Filter
+{
+    /**
+     * Field is null.
+     */
+    public const TYPE_EMPTY = 1;
+
+    /**
+     *  Field is not null.
+     */
+    public const TYPE_NOT_EMPTY = 2;
+
+    /**
+     * @param string            $alias
+     * @param string            $field
+     * @param array|string|null $data
+     */
+    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data): void
+    {
+        if (null === $data || !\is_array($data) || !\array_key_exists('value', $data)) {
+            return;
+        }
+
+        if ((int) $data['value'] === static::TYPE_NOT_EMPTY) {
+            $queryBuilder->andWhere(sprintf('%s.%s IS null', $alias, $field));
```

Comment:
If I filter by TYPE_NOT_EMPTY, I expect the not null

Reply:
And would it possible to handle OTM, OTO, MTO, ... relations ?

If I have A human entity OTM with dog entity.
If I look for every human without dog (i.e. human.dogs.id null), or every dog with human without a name (i.e dog.human.name), could this filter be used ?